seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java

```diff
@@ -106,5 +118,68 @@ public List<String> getFileNamesByPath(HadoopConf hadoopConf, String path) throw
     @Override
     public void setPluginConfig(Config pluginConfig) {
         this.pluginConfig = pluginConfig;
+        if (pluginConfig.hasPath(BaseSourceConfig.PARSE_PARTITION_FROM_PATH)) {
+            isMergePartition = pluginConfig.getBoolean(BaseSourceConfig.PARSE_PARTITION_FROM_PATH);
+        }
+    }
+
+    @Override
+    public SeaTunnelRowType getActualSeaTunnelRowTypeInfo() {
+        return isMergePartition ? seaTunnelRowTypeWithPartition : seaTunnelRowType;
+    }
+
+    protected Map<String, String> parsePartitionsByPath(String path) {
+        LinkedHashMap<String, String> partitions = new LinkedHashMap<>();
+        Arrays.stream(path.split("/", -1))
+                .filter(split -> split.contains("="))
+                .map(split -> split.split("=", -1))
+                .forEach(kv -> partitions.put(kv[0], kv[1]));
+        return partitions;
+    }
+
+    protected SeaTunnelRow mergePartitionFields(String path, SeaTunnelRow seaTunnelRow) {
+        Map<String, String> partitionsMap = parsePartitionsByPath(path);
```

Comment:
Each row will call this method `mergePartitionFields ` , we must reduce the execution time of this method. 
`parsePartitionsByPath` is not necessary to execute every row. It can do once when we open the file.